JavaScript/JavaScriptSDK/Telemetry/PageViewPerformance.ts

```diff
@@ -137,12 +146,31 @@ module Microsoft.ApplicationInsights.Telemetry {
         }
 
         public static getDuration(start: any, end: any): number {
-            var duration = 0;
+            var duration = undefined;
             if (!(isNaN(start) || isNaN(end))) {
                 duration = Math.max(end - start, 0);
             }
 
             return duration;
         }
+
+        /**
+         * GoogleBot is returning invalid values in performance.timing API.
+         * This method tells if given durations should be excluded from collection.
+         */
+        public static shouldCollectDuration(...durations: number[]): boolean {
+            let userAgent = navigator.userAgent;
+            let isGoogleBot = userAgent ? userAgent.toLowerCase().indexOf("googlebot") !== -1 : false;
+
+            if (isGoogleBot) {
```

Comment:
Shouldn't this be an or condition?  It is GoogleBot, or it is > MAX_DURATION_ALLOWED?
Maybe GoogleBot pretends to be IOS and still reports 3 hours.